src/consumeroffset/kafka_offset_collect_test.go

```diff
@@ -1 +1,203 @@
+	case 0:
+		return 25, nil
+	case 1:
+		return 30, nil
+	}
+	return 0, nil
+}
+
+func TestCollectOffsetsForConsumerGroup(t *testing.T) { // nolint: funlen
+	// MemberAssignment mock created as in sarama's consumer_group_member_test.go
+	members := map[string]*sarama.GroupMemberDescription{
+		clientID: {
+			ClientId:       clientID,
+			ClientHost:     "a-host",
+			MemberMetadata: nil,
+			MemberAssignment: []byte{
+				0, 0, // Version
+				0, 0, 0, 1, // Topic array length
+				0, 3, 'o', 'n', 'e', // Topic one
+				0, 0, 0, 3, // Topic one, partition array length
+				0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 4, // 0, 2, 4
+				0, 0, 0, 3, 0x01, 0x02, 0x03, // Userdata
+			},
+		},
+	}
+
+	args.GlobalArgs = &args.ParsedArguments{}
+
+	testCases := []struct {
+		name                        string
+		inactiveConsumerGroupOffset bool
+		consumerGroupOffsetByTopic  bool
+		consumerGroup               string
+		cGroupEntities              map[string]map[string]float64
+		topicEntities               map[string]map[string]float64
+		numEntites                  int
+	}{
+		{
+			name:                        "Only active consumers",
+			inactiveConsumerGroupOffset: false,
+			consumerGroupOffsetByTopic:  false,
+			consumerGroup:               consumerGroupOne,
+			cGroupEntities: map[string]map[string]float64{
+				consumerGroupOne: {
+					// comes from: 25 - 10 + 30 - 10
+					"totalLag": 35,
+					// comes from max: 30 - 10
+					"maxLag":          20,
+					"activeConsumers": 1,
+				},
+			},
+			topicEntities: nil,
+			numEntites:    4,
+		},
+		{
+			name:                        "With inactive consumers",
+			inactiveConsumerGroupOffset: true,
+			consumerGroupOffsetByTopic:  false,
+			consumerGroup:               consumerGroupOne,
+			cGroupEntities: map[string]map[string]float64{
+				consumerGroupOne: {
+					// comes from: (25 - 10 + 30 - 10) + (25 - 10 + 30 - 10)
+					"totalLag": 70,
+					// comes from max: 30 - 10
+					"maxLag":          20,
+					"activeConsumers": 1,
+				},
+			},
+			topicEntities: nil,
+			numEntites:    4,
+		},
+		{
+			name:                        "With inactive consumers and topic aggregation",
+			inactiveConsumerGroupOffset: true,
+			consumerGroupOffsetByTopic:  true,
+			consumerGroup:               consumerGroupOne,
+			cGroupEntities: map[string]map[string]float64{
+				consumerGroupOne: {
+					// comes from: (25 - 10 + 30 - 10) + (25 - 10 + 30 - 10)
+					"totalLag": 70,
+					// comes from max: 30 - 10
+					"maxLag":          20,
+					"activeConsumers": 1,
+				},
+			},
+			topicEntities: map[string]map[string]float64{
+				topicOne: {
+					// comes from: 25 - 10 + 30 - 10
+					"totalLag": 35,
+					// comes from max: 30 - 10
+					"maxLag":          20,
+					"activeConsumers": 1,
+				},
+				topicTwo: {
+					// comes from: 25 - 10 + 30 - 10
+					"totalLag": 35,
+					// comes from max: 30 - 10
+					"maxLag":          20,
+					"activeConsumers": 0,
+				},
+			},
+			numEntites: 6,
+		},
```

Comment:
would it make sense to add also a case for `inactiveConsumerGroupOffset: false` and `consumerGroupOffsetByTopic: true`?